patches/server/0726-Expose-firework-item-directly-manually-setting-fligh.patch

```diff
@@ -0,0 +1,126 @@
++        // Paper Start - Expose firework item directly
++//        ItemStack item = this.getHandle().getEntityData().get(FireworkRocketEntity.DATA_ID_FIREWORKS_ITEM);
++//
++//        if (item.isEmpty()) {
++//            item = new ItemStack(Items.FIREWORK_ROCKET);
++//            this.getHandle().getEntityData().set(FireworkRocketEntity.DATA_ID_FIREWORKS_ITEM, item);
++//        }
++//
++//        this.item = CraftItemStack.asCraftMirror(item);
++//
++//        // Ensure the item is a firework...
++//        if (this.item.getType() != Material.FIREWORK_ROCKET) {
++//            this.item.setType(Material.FIREWORK_ROCKET);
++//        }
++        // Paper End - Expose firework item directly
+     }
+ 
+     @Override
+@@ -51,17 +53,28 @@ public class CraftFirework extends CraftProjectile implements Firework {
+ 
+     @Override
+     public FireworkMeta getFireworkMeta() {
+-        return (FireworkMeta) this.item.getItemMeta();
++        // Paper Start - Expose firework item directly
++        org.bukkit.inventory.ItemStack itemStack = getItem();
++        if (itemStack.getType() != Material.FIREWORK_ROCKET) {
++            return (FireworkMeta) org.bukkit.craftbukkit.inventory.CraftItemFactory.instance().getItemMeta(Material.FIREWORK_ROCKET); // Return normal meta
++        }
++        return (FireworkMeta) itemStack.getItemMeta();
++        // Paper End - Expose firework item directly
+     }
+ 
+     @Override
+     public void setFireworkMeta(FireworkMeta meta) {
+-        this.item.setItemMeta(meta);
+-
++        // Paper Start - Expose firework item directly
++        org.bukkit.inventory.ItemStack itemStack = getItem();
++        if (itemStack.getType() != Material.FIREWORK_ROCKET) {
++            itemStack.setType(Material.FIREWORK_ROCKET);
++        }
++        itemStack.setItemMeta(meta);
++        // Paper End - Expose firework item directly
+         // Copied from EntityFireworks constructor, update firework lifetime/power
+         this.getHandle().lifetime = 10 * (1 + meta.getPower()) + this.random.nextInt(6) + this.random.nextInt(7);
+ 
+-        this.getHandle().getEntityData().markDirty(FireworkRocketEntity.DATA_ID_FIREWORKS_ITEM);
++        setItem(itemStack); // Paper - Expose firework item directly
+     }
+ 
+     @Override
+@@ -91,4 +104,41 @@ public class CraftFirework extends CraftProjectile implements Firework {
+         return boostedEntity != null ? (org.bukkit.entity.LivingEntity) boostedEntity.getBukkitEntity() : null;
+     }
+     // Paper end
++    // Paper start - Expose firework item directly + manually setting flight
++    @Override
++    public org.bukkit.inventory.ItemStack getItem() {
++        ItemStack itemStack = getHandle().getEntityData().get(FireworkRocketEntity.DATA_ID_FIREWORKS_ITEM);
++
++        if (itemStack.isEmpty()) {
++            return new org.bukkit.inventory.ItemStack(Material.FIREWORK_ROCKET);
++        } else {
++            return CraftItemStack.asBukkitCopy(itemStack);
++        }
```

Comment:
Theres a getItem method on FireworkRocketEntity that does this logic for you, can't you just wrap that in a asBukkitCopy?